src/librustdoc/clean/inline.rs

```diff
@@ -229,10 +229,8 @@ pub(crate) fn item_relative_path(tcx: TyCtxt<'_>, def_id: DefId) -> Vec<Symbol>
 /// These names are used later on by HTML rendering to generate things like
 /// source links back to the original item.
 pub(crate) fn record_extern_fqn(cx: &mut DocContext<'_>, did: DefId, kind: ItemType) {
-    if did.is_local() {
-        if cx.cache.exact_paths.contains_key(&did) {
-            return;
-        }
+    if did.is_local() && cx.cache.exact_paths.contains_key(&did) {
```

Comment:
It's not the same condition anymore here.

Reply:
Ok so in here, I think the code will be a bit slower because if the whole condition is false, then it'll run the next condition even if the `did` is local. I'm sure it's close to nothing but I think it still worth keeping as it was.

Reply:
done